packages/core/lib/command.js

```diff
@@ -107,7 +108,12 @@ class Command {
     if (!Array.isArray(inputStrings)) inputStrings = inputStrings.split(" ");
     // Method `extractFlags(args)` : Extracts the `--option` flags from arguments
     const inputOptions = extractFlags(inputStrings);
-    const validOptions = result.command.help.options
+
+    //adding allowed global options as enumerated in each command
+    const allowedGlobalOptions = result.command.help.allowedGlobalOptions.map(tag => commandOptions[tag].option ? commandOptions[tag] : "");
```

Comment:
Better to avoid use of `""` where it's not necessary:
```suggestion
    const allowedGlobalOptions = result.command.help.allowedGlobalOptions
      .filter(tag => tag in commandOptions)
      .map(tag => commandOptions[tag].option);
```

This will mean that Truffle ignores malformed commands and falls back on the existing warning mechanism for unknown flags, which seems to be what you want.

Reply:
I just pushed a commit switching this out for `filter` as you suggested. What you have here doesn't work because we don't want to return the short option, but the object attached to it. Let me know if that doesn't make sense!